### PR TITLE
E2E for TensorFlow Integration

### PR DESCRIPTION
**Pull request by thandayuthapani**
This PR involves E2E for TensorFlow Integration with volcano

### CODE REVIEW

**Comment by TommyLike** [comment]
@thandayuthapani  Download every images and run different samples for every application running on volcano would slow down our developing process. Maybe we should setup up a cronjob for this kind of e2e tests?

cc @k82cn  @hzxuzhonghu

File: test/e2e/util.go [hunk 1]
```diff
@@ -645,7 +646,7 @@ func waitJobStateAborted(ctx *context, job *vkv1.Job) error {
 
 func waitJobPhaseExpect(ctx *context, job *vkv1.Job, state vkv1.JobPhase) error {
 	var additionalError error
-	err := wait.Poll(100*time.Millisecond, oneMinute, func() (bool, error) {
+	err := wait.Poll(100*time.Millisecond, twoMinute, func() (bool, error) {
```
Comment (hzxuzhonghu) [inline test/e2e/util.go]: this may increase total test time, I would like to add a parameter to `waitJobPhaseExpect`

**Comment by hzxuzhonghu** [comment]
I think this is good, we should have a tf job case. Cronjob does not fit very much, it can not prevent a pr from breaking cases like tf job.

For the speed issue, the docker images should be reusable.

File: test/e2e/util.go (hunk 1 above)
Comment (thandayuthapani) [reply test/e2e/util.go]: Updated

**Comment by thandayuthapani** [comment]
> @thandayuthapani Download every images and run different samples for every application running on volcano would slow down our developing process. Maybe we should setup up a cronjob for this kind of e2e tests?
> 
> cc @k82cn @hzxuzhonghu

@TommyLike To my Knowledge downloading Images will not affect test time much, since they provide high speed internet, which is around 35MB/s. So downloading images might add 10 more secs in E2E

**Comment by TommyLike** [comment]
> > @thandayuthapani Download every images and run different samples for every application running on volcano would slow down our developing process. Maybe we should setup up a cronjob for this kind of e2e tests?
> > cc @k82cn @hzxuzhonghu
> 
> @TommyLike To my Knowledge downloading Images will not affect test time much, since they provide high speed internet, which is around 35MB/s. So downloading images might add 10 more secs in E2E

We need to consider how much time it takes to download the image  and how much time it takes to load the images into kind and to complete the e2e tests.
```
⚡ root@husheng-test  ~  docker image ls | grep ed417b28e09c
thanda/tf-operator-example    1.0                                        ed417b28e09c        42 hours ago        1.25GB
```

```
 ⚡ root@husheng-test  ~  kind load docker-image thanda/tf-operator-example:1.0  --name integration
Execution time: 0h:02m:57s sec
```

image

**Comment by thandayuthapani** [comment]
> We need to consider how much time it takes to download the image and how much time it takes to load the images into kind and to complete the e2e tests.
> 
> ```
> ⚡ root@husheng-test  ~  docker image ls | grep ed417b28e09c
> thanda/tf-operator-example    1.0                                        ed417b28e09c        42 hours ago        1.25GB
> ```
> 
> ```
>  ⚡ root@husheng-test  ~  kind load docker-image thanda/tf-operator-example:1.0  --name integration
> Execution time: 0h:02m:57s sec
> ```
> 
> image

1.25GB Image which has been posted is old one, Have updated that image with lighter one.
```
docker images | grep thanda/tf-operator-example
thanda/tf-operator-example    1.0                 1bdc930499c4        4 hours ago         461MB
```
And time to load docker-image in my local setup is 
```
time kind load docker-image thanda/tf-operator-example:1.0 --name integration

real	0m22.613s
```
For E2E test it will take around 3 mins to complete but that is the simplest tensorflow training example I could find.

**Comment by TommyLike** [comment]
@thandayuthapani thanks for your update, and that's why I am thinking if we need considering move this into another job

**Comment by thandayuthapani** [comment]
@TommyLike Have reduced number of train steps, now the test runs in CI for 78 Secs, if that is okay we can maintain that, else we can reduce steps further also if necessary.
Screenshot from 2019-07-24 16-43-27

**Comment by hzxuzhonghu** [comment]
@thandayuthapani One point, the image used should be in volcanosh instead of your own repo.

**Comment by thandayuthapani** [comment]
> @thandayuthapani One point, the image used should be in volcanosh instead of your own repo.

Sure will move that to volcanosh repo

**Comment by thandayuthapani** [comment]
> @thandayuthapani One point, the image used should be in volcanosh instead of your own repo.

Have updated the same.

File: example/integrations/tensorflow/dist-mnist/tf-dist-mnist-example.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          containers:
+            - command:
+                - sh
+                - -c
+                - |
+                  PS_HOST=`cat /etc/volcano/ps.host | sed 's/$/&:2222/g' | sed 's/^/"/;s/$/"/' | tr "\n" ","`;
+                  WORKER_HOST=`cat /etc/volcano/worker.host | sed 's/$/&:2222/g' | sed 's/^/"/;s/$/"/' | tr "\n" ","`;
+                  export TF_CONFIG={\"cluster\":{\"ps\":[${PS_HOST}],\"worker\":[${WORKER_HOST}]},\"task\":{\"type\":\"ps\",\"index\":${VK_TASK_INDEX}},\"environment\":\"cloud\"};
+                  python /var/tf_dist_mnist/dist_mnist.py
+              image: thanda/tf-operator-example:1.0
```
Comment (TommyLike) [inline example/integrations/tensorflow/dist-mnist/tf-dist-mnist-example.yaml]: volcanosh/dist-mnist-tf-example:0.0.1

File: example/integrations/tensorflow/dist-mnist/tf-dist-mnist-example.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          containers:
+            - command:
+                - sh
+                - -c
+                - |
+                  PS_HOST=`cat /etc/volcano/ps.host | sed 's/$/&:2222/g' | sed 's/^/"/;s/$/"/' | tr "\n" ","`;
+                  WORKER_HOST=`cat /etc/volcano/worker.host | sed 's/$/&:2222/g' | sed 's/^/"/;s/$/"/' | tr "\n" ","`;
+                  export TF_CONFIG={\"cluster\":{\"ps\":[${PS_HOST}],\"worker\":[${WORKER_HOST}]},\"task\":{\"type\":\"worker\",\"index\":${VK_TASK_INDEX}},\"environment\":\"cloud\"};
+                  python /var/tf_dist_mnist/dist_mnist.py
+              image: thanda/tf-operator-example:1.0
```
Comment (TommyLike) [inline example/integrations/tensorflow/dist-mnist/tf-dist-mnist-example.yaml]: ditto

File: example/integrations/tensorflow/dist-mnist/tf-dist-mnist-example.yaml (hunk 2 above)
Comment (thandayuthapani) [reply example/integrations/tensorflow/dist-mnist/tf-dist-mnist-example.yaml]: updated

File: example/integrations/tensorflow/dist-mnist/tf-dist-mnist-example.yaml (hunk 3 above)
Comment (thandayuthapani) [reply example/integrations/tensorflow/dist-mnist/tf-dist-mnist-example.yaml]: updated

**Comment by thandayuthapani** [comment]
@k82cn Can you please retrigger this build.

**Comment by thandayuthapani** [comment]
@k82cn @TommyLike Please have a look

**Pull request merged**